File: Web/src/api-services/models/menu-output.ts
/* tslint:disable */
/* eslint-disable */

import { MenuOutput } from './menu-output';
import { MenuTypeEnum } from './menu-type-enum';
import { StatusEnum } from './status-enum';
import { SysMenuMeta } from './sys-menu-meta';
/**
 * 系统菜单返回结果
 * @export
 * @interface MenuOutput
 */
export interface MenuOutput {
    /**
     * Id
     * @type {number}
     * @memberof MenuOutput
     */
    id?: number;
    /**
     * 父Id
     * @type {number}
     * @memberof MenuOutput
     */
    pid?: number;
    /**
     * 
     * @type {MenuTypeEnum}
     * @memberof MenuOutput
     */
    type?: MenuTypeEnum;
    /**
     * 名称
     * @type {string}
     * @memberof MenuOutput
     */
    name?: string | null;
    /**
     * 路由地址
     * @type {string}
     * @memberof MenuOutput
     */
    path?: string | null;
    /**
     * 组件路径
     * @type {string}
     * @memberof MenuOutput
     */
    component?: string | null;
    /**
     * 权限标识
     * @type {string}
     * @memberof MenuOutput
     */
    permission?: string | null;
    /**
     * 重定向
     * @type {string}
     * @memberof MenuOutput
     */
    redirect?: string | null;
    /**
     * 排序
     * @type {number}
     * @memberof MenuOutput
     */
    orderNo?: number;
    /**
     * 
     * @type {StatusEnum}
     * @memberof MenuOutput
     */
    status?: StatusEnum;
    /**
     * 备注
     * @type {string}
     * @memberof MenuOutput
     */
    remark?: string | null;
    /**
     * 
     * @type {SysMenuMeta}
     * @memberof MenuOutput
     */
    meta?: SysMenuMeta;
    /**
     * 菜单子项
     * @type {Array<MenuOutput>}
     * @memberof MenuOutput
     */
    children?: Array<MenuOutput> | null;
}

File: Admin.NET/Admin.NET.Core/Service/Menu/Dto/MenuOutput.cs
﻿namespace Admin.NET.Core.Service;

/// <summary>
/// 系统菜单返回结果
/// </summary>
public class MenuOutput
{
    /// <summary>
    /// Id
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// 父Id
    /// </summary>
    public long Pid { get; set; }

    /// <summary>
    /// 菜单类型（0目录 1菜单 2按钮）
    /// </summary>
    public MenuTypeEnum Type { get; set; }

    /// <summary>
    /// 名称
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// 路由地址
    /// </summary>
    public string Path { get; set; }

    /// <summary>
    /// 组件路径
    /// </summary>
    public string Component { get; set; }

    /// <summary>
    /// 权限标识
    /// </summary>
    public string Permission { get; set; }

    /// <summary>
    /// 重定向
    /// </summary>
    public string Redirect { get; set; }

    /// <summary>
    /// 排序
    /// </summary>
    public int OrderNo { get; set; }

    /// <summary>
    /// 状态
    /// </summary>
    public StatusEnum Status { get; set; }

    /// <summary>
    /// 备注
    /// </summary>
    public string Remark { get; set; }

    /// <summary>
    /// 菜单Meta
    /// </summary>
    public SysMenuMeta Meta { get; set; }

    /// <summary>
    /// 菜单子项
    /// </summary>
    public List<MenuOutput> Children { get; set; }
}

/// <summary>
/// 菜单Meta配置
/// </summary>
public class SysMenuMeta
{
    /// <summary>
    /// 标题
    /// </summary>
    public string Title { get; set; }

    /// <summary>
    /// 图标
    /// </summary>
    public string Icon { get; set; }

    /// <summary>
    /// 是否内嵌
    /// </summary>
    public bool IsIframe { get; set; }

    /// <summary>
    /// 外链链接
    /// </summary>
    public string IsLink { get; set; }

    /// <summary>
    /// 是否隐藏
    /// </summary>
    public bool IsHide { get; set; }

    /// <summary>
    /// 是否缓存
    /// </summary>
    public bool IsKeepAlive { get; set; }

    /// <summary>
    /// 是否固定
    /// </summary>
    public bool IsAffix { get; set; }
}

/// <summary>
/// 配置菜单对象映射
/// </summary>
public class SysMenuMapper : IRegister
{
    public void Register(TypeAdapterConfig config)
    {
        config.ForType<SysMenu, MenuOutput>()
            .Map(t => t.Meta.Title, o => o.Title)
            .Map(t => t.Meta.Icon, o => o.Icon)
            .Map(t => t.Meta.IsIframe, o => o.IsIframe)
            .Map(t => t.Meta.IsLink, o => o.OutLink)
            .Map(t => t.Meta.IsHide, o => o.IsHide)
            .Map(t => t.Meta.IsKeepAlive, o => o.IsKeepAlive)
            .Map(t => t.Meta.IsAffix, o => o.IsAffix);
    }
}